proxy/envoy/discovery.go

```diff
@@ -405,10 +405,6 @@ func (ds *DiscoveryService) ListEndpoints(request *restful.Request, response *re
 }
 
 func (ds *DiscoveryService) parseRole(request *restful.Request) (proxy.Node, error) {
-	if sc := request.PathParameter(ServiceCluster); sc != ds.Mesh.IstioServiceCluster {
-		return proxy.Node{}, fmt.Errorf("unexpected %s %q", ServiceCluster, sc)
-	}
-
```

Comment:
cc @andraxylia you might want to remove these lines from your PR to avoid merge conflicts. This is what I was telling you the other day.